operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -19,47 +20,92 @@ import (
 
 var hashSeparator = []byte(",")
 
-func getSecret(ctx context.Context, k k8s.Client, stack *lokiv1.LokiStack) (*corev1.Secret, error) {
-	var storageSecret corev1.Secret
+func getSecrets(ctx context.Context, k k8s.Client, stack *lokiv1.LokiStack, fg configv1.FeatureGates) (*corev1.Secret, *corev1.Secret, error) {
+	var (
+		storageSecret     corev1.Secret
+		managedAuthSecret corev1.Secret
+	)
+
 	key := client.ObjectKey{Name: stack.Spec.Storage.Secret.Name, Namespace: stack.Namespace}
 	if err := k.Get(ctx, key, &storageSecret); err != nil {
 		if apierrors.IsNotFound(err) {
-			return nil, &status.DegradedError{
+			return nil, nil, &status.DegradedError{
 				Message: "Missing object storage secret",
 				Reason:  lokiv1.ReasonMissingObjectStorageSecret,
 				Requeue: false,
 			}
 		}
-		return nil, kverrors.Wrap(err, "failed to lookup lokistack storage secret", "name", key)
+		return nil, nil, kverrors.Wrap(err, "failed to lookup lokistack storage secret", "name", key)
+	}
+
+	if fg.OpenShift.ManagedAuthEnv {
+		secretName, ok := stack.Annotations[storage.AnnotationCredentialsRequestsSecretRef]
+		if !ok {
+			return nil, nil, &status.DegradedError{
+				Message: "Missing OpenShift cloud credentials request",
+				Reason:  lokiv1.ReasonMissingCredentialsRequest,
+				Requeue: true,
+			}
+		}
+
+		managedAuthCredsKey := client.ObjectKey{Name: secretName, Namespace: stack.Namespace}
+		if err := k.Get(ctx, managedAuthCredsKey, &managedAuthSecret); err != nil {
+			if apierrors.IsNotFound(err) {
+				return nil, nil, &status.DegradedError{
+					Message: "Missing OpenShift cloud credentials secret",
+					Reason:  lokiv1.ReasonMissingManagedAuthSecret,
+					Requeue: true,
+				}
+			}
+			return nil, nil, kverrors.Wrap(err, "failed to lookup OpenShift CCO managed authentication credentials secret", "name", stack)
+		}
+
+		return &storageSecret, &managedAuthSecret, nil
 	}
 
-	return &storageSecret, nil
+	return &storageSecret, nil, nil
 }
 
-// extractSecret reads a k8s secret into a manifest object storage struct if valid.
-func extractSecret(s *corev1.Secret, secretType lokiv1.ObjectStorageSecretType) (storage.Options, error) {
-	hash, err := hashSecretData(s)
+// extractSecrets reads a k8s secret into a manifest object storage struct if valid.
```

Comment:
maybe my suggestion is not perfect but something of the sorts
```suggestion
// extractSecrets reads the k8s obj storage secret into a manifest object storage struct if valid.
// The managed auth is also read into the manifest object under the right circumstances.
```